boards/adafruit/qt_py_esp32s3/adafruit_qt_py_esp32s3_procpu_psram.dts

```diff
@@ -0,0 +1,12 @@
+/*
+ * Copyright (c) 2024 Ian Wakely
+ */
+
+/dts-v1/;
```

Comment:
For my own understanding, you are explicitly deleting this node to make sure it gets replaced by the one from espressif/esp32s3/esp32s3_wroom_n4r2.dtsi? Do you need to do this with normal dts inclusions?

Reply:
By having a board without a board revision, it will not have an overlay file and will only load the dts configuration from the .dts file, which is a problem as you need to have a different file included for each revision, so the workaround is allow the original be included, then delete what the original sets, and include the one that is wanted

Reply:
I also probably want to delete the `psram0` node as well.

Reply:
> The overlay probably shouldn't be including the 'dtsi' files I had, right?

Correct, they will get it from the base .dts file

> I also probably want to delete the psram0 node as well.

I don't see that, the .overlay file is for the psnode one so it should have it defined

Reply:
> The overlay probably shouldn't be including the 'dtsi' files I had, right?

Nope, they are very much required. Doesn't seem to build without them.

Reply:
> > The overlay probably shouldn't be including the 'dtsi' files I had, right?
> 
> Correct, they will get it from the base .dts file

When removing that include, I get the following device tree error:

```
devicetree error: /chosen: undefined node label 'boot_partition'
```

CMake is reporting that BOARD.dts is adafruit_qt_py_esp32s3_procpu.dts (which is correct).
It then shows the devicetree overlay as adafruit_qt_py_esp32s3_procpu_psram.overlay and app.overlay from mcuboot.

Reply:
I was able to get past the boot partition issue by re-including "espressif/partitions_0x0_amp.dtsi" in the overlay, but it then wasn't able to find a console driver as it believed that all of the espressif,esp32-uart nodes were missing.

I'm starting to think there might be a limitation of how dtc and the scripts that run it are used, and we're hitting that limitation by not re-including the common dtsi file.

What I have pushed at least builds, but not as confident that it is as correct as the original "variant" implementation I originally had.

Reply:
@raveious try to move the `#include <espressif/partitions_0x0_amp.dtsi>` after other includes as it referencing the `flash0` node.

Reply:
> @raveious try to move the `#include <espressif/partitions_0x0_amp.dtsi>` after other includes as it referencing the `flash0` node.

I tried putting it at the end of the includes in the device tree with no effect. When putting at the end of the includes for the overlay, it makes it a bit further but fails with the following series of errors:
```
warning: UART_CONSOLE (defined at drivers/console/Kconfig:42) was assigned the value 'y' but got the
value 'n'.
<snip>
CMake Warning at <snip>/zephyr/CMakeLists.txt:957 (message):
  No SOURCES given to Zephyr library: drivers__console

  Excluding target from build.


CMake Warning at <snip>/zephyr/CMakeLists.txt:957 (message):
  No SOURCES given to Zephyr library: drivers__serial

  Excluding target from build.


CMake Warning at <snip>/zephyr/CMakeLists.txt:957 (message):
  No SOURCES given to Zephyr library: drivers__spi

  Excluding target from build.
```
Ultimately failing with a undefined reference to `bootloader_console_init`. This seems to be because `DT_HAS_ESPRESSIF_ESP32_UART_ENABLED` is missing from the generated device tree header. That define gets setup when it finds `espressif,esp32-uart` which is defined in `dts/xtensa/espressif/esp32s3/esp32s3_common.dtsi` and enabled in `adafruit_qt_py_esp32s3_procpu.dts` (or *_common.dtsi if you are looking at this PR). However, it seems to be ignored even though I see if being sucked into the build.

I've pushed the commits to [here](https://github.com/raveious/zephyr/commit/291d87a7d0bc8567d21894fdf33d76e716bcdba4) and I've been building with one of the following commands:
```
west build -b adafruit_qt_py_esp32s3@psram/esp32s3/procpu --sysbuild samples/drivers/led/led_strip
west build -b adafruit_qt_py_esp32s3@psram/esp32s3/procpu --sysbuild samples/hello_world
```

Reply:
Apply this patch, should be fine after: https://pastebin.com/Z113TaMk

Reply:
> Apply this patch, should be fine after: https://pastebin.com/Z113TaMk

That patch does fix the build. It does seem odd that this has to be copied from the SOC dtsi files though. Won't this cause maintenance issues in the future by having snippets like this all over the place?

Reply:
So essentially this is a hack, board revisions are used for textual/numerical board revisions, having a board revision where one is empty is a special case for this board because of how the manufacturer has decided to produce a badly labelled board

Reply:
> So essentially this is a hack, board revisions are used for textual/numerical board revisions, having a board revision where one is empty is a special case for this board because of how the manufacturer has decided to produce a badly labelled board

I completely agree that every board should have a revision on it. I guess my question is, knowing how much of a hack this feels, was this better or worse than the variants approach?